validphys2/src/validphys/n3fit_data.py

```diff
@@ -92,7 +92,7 @@ def tr_masks(data, replica_trvlseed):
         cuts = dataset.cuts
         ndata = len(cuts.load()) if cuts else dataset.commondata.ndata
         frac = dataset.frac
-        trmax = int(frac * ndata)
+        trmax = (np.random.rand() < ndat * frac - int(ndat * frac)) + int(ndat * frac)
```

Comment:
```suggestion
        trmax = (np.random.rand() < ndata * frac - int(ndata * frac)) + int(ndata * frac)
```

ups. Also, now there is the problem that the regression tests (should) fail but oh well. I'll update them next week